content/src/ports/denylist.ts

```diff
@@ -1,50 +1,104 @@
+import { IBaseComponent } from '@well-known-components/interfaces'
 import { resolve } from 'path'
 import { createInterface } from 'readline'
+import { URL } from 'url'
 import { EnvironmentConfig } from '../Environment'
 import { AppComponents } from '../types'
+import { FSComponent } from './fs'
 
-export interface DenylistComponent {
-  isDenyListed(hash: string): boolean
+export interface Denylist {
+  isDenylisted(id: string): boolean
 }
 
-export async function createDenylistComponent(
-  components: Pick<AppComponents, 'env' | 'logs' | 'fs'>
-): Promise<DenylistComponent> {
-  const logs = components.logs.getLogger('Denylist')
-  const bannedList = new Set()
-
+export async function createDenylist(
+  components: Pick<AppComponents, 'env' | 'logs' | 'fetcher'> & {
+    fs: Pick<FSComponent, 'createReadStream' | 'existPath'>
+  }
+): Promise<Denylist & IBaseComponent> {
+  const logger = components.logs.getLogger('Denylist')
+  const deniedContentIdentifiers = new Set()
   const fileName = resolve(
     components.env.getConfig(EnvironmentConfig.STORAGE_ROOT_FOLDER),
     components.env.getConfig(EnvironmentConfig.DENYLIST_FILE_NAME)
   )
 
-  try {
-    await components.fs.existPath(fileName)
-
-    const content = components.fs.createReadStream(fileName, {
-      encoding: 'utf-8'
-    })
-
-    const lines = createInterface({
-      input: content,
-      crlfDelay: Infinity
+  const denylistsUrlsRaw = components.env.getConfig<string>(EnvironmentConfig.DENYLIST_URLS)?.split(/[\r\n\s]+/gm) ?? []
+  const denylistsUrls: URL[] = denylistsUrlsRaw
+    .filter((url) => {
+      try {
+        new URL(url)
+        return true
+      } catch {
+        return false
+      }
     })
+    .map((url) => new URL(url))
+  logger.info(`File: ${fileName}`)
```

Comment:
Can we make these logs more descriptive about what are they printing?

Reply:
sure